docs/model.md

```diff
@@ -0,0 +1,61 @@
+## Constraint Programming Model
+
+version = v1.0.0
+
+### Input data
+
+- $`Nb\_Zones \subset \mathbf{N}`$ : The number of zones
+- $`Nb\_Trains \subset \mathbf{N}`$ : The number of trains
+- $`Nb\_Steps \subset \mathbf{N}`$ : The number of steps
+
+
+
+- $`train_s \in \{1..Nb\_Trains\}, \forall s \in \{1..Nb\_Steps\}`$ : The train associated with the step s
+- $`zone_s \in \{1..Nb\_Zones\}, \forall s \in \{1..Nb\_Steps\}`$ : The zone associated with the step s
+- $`prev_s \in \{0..Nb\_Steps\}, \forall s \in \{1..Nb\_Steps\}`$ : The step preceding s (0 if None)
+
+
+- $`min\_arrival_s \in \mathbf{N}, \forall s \in \{1..Nb\_Steps\}`$ : The min arrival time of a step
+- $`min\_departure_s \in \mathbf{N}, \forall s \in \{1..Nb\_Steps\}`$ : The min departure time of a step
+- $`min\_duration_s \in \mathbf{N}, \forall s \in \{1..Nb\_Steps\}`$ : The min duration of a step
+
+- $`is\_fixed_s \in \{True, False\}, \forall s \in \{1..Nb\_Steps\}`$ : If the duration of a step cannot be changed
+
+### Decision variables
+
+- $`arrival_s \in \mathbf{N}, \forall s \in \{1..Nb\_Steps\}`$ : The arrival time of a step
+- $`departure_s \in \mathbf{N}, \forall s \in \{1..Nb\_Steps\}`$ : The departure time of a step
+
+### Constraints
+
+- Two trains shouldn't collide
+
+$`\forall s1, s2 \in \{1..Nb\_Steps\} \; s.t. \; s1 < s2, arrival_{s1} \geq departure_{s2} \lor departure_{s1} \leq arrival_{s2}`$
+
+- A step cannot overtake its reference
+
```

Comment:
$$\left\\{
\begin{array}{ll}
\forall s \in \\{1,..,N_{steps}\\}, &  arrival_s \geq min\textunderscore arrival_s\\
\forall s \in \\{1,..,N_{steps}\\}, & departure_s \geq min\textunderscore departure_s\\
\end{array}
\right.
$$

**Les noms de variables avec un _ ne sont pas une bonne idée en LaTeX !!**